api/src/test/java/org/apache/iceberg/transforms/TestResiduals.java

```diff
@@ -156,4 +158,46 @@ public void testUnpartitionedResiduals() {
           expr, residualEvaluator.residualFor(Row.of()));
     }
   }
+
+  @Test
+  public void testIn() {
+    Schema schema = new Schema(
+        Types.NestedField.optional(50, "dateint", Types.IntegerType.get()),
+        Types.NestedField.optional(51, "hour", Types.IntegerType.get())
+    );
+
+    PartitionSpec spec = PartitionSpec.builderFor(schema)
+        .identity("dateint")
+        .build();
+
+    ResidualEvaluator resEval = ResidualEvaluator.of(spec,
+        in("dateint", 20170815, 20170816, 20170817), true);
+
+    Expression residual = resEval.residualFor(Row.of(20170815));
+    Assert.assertEquals("Residual should be alwaysTrue", alwaysTrue(), residual);
+
+    residual = resEval.residualFor(Row.of(20180815));
+    Assert.assertEquals("Residual should be alwaysFalse", alwaysFalse(), residual);
+  }
+
+  @Test
+  public void testNotIn() {
```

Comment:
Can you also add tests for `in` and `notIn` with a timestamp (`ts` column) to date transform (`day`)?

For the `in` case, the residual for a date that contains one of the timestamps should be the original `in` predicate. If the date doesn't match one of the `in` predicate timestamp values, then it should be `alwaysFalse`. For `notIn`, if the date matches it should be the original `notIn` predicate and if it does not match, it should be `alwaysTrue`.